content/blog/2018/08/2018-08-14-code-coverage-api-plugin-1.0-release.adoc

```diff
@@ -0,0 +1,198 @@
+
+=== Overview
+
+Code Coverage API plugin is one of https://jenkins.io/projects/gsoc/[GSoC 2018 Jenkins projects].
+
+There are a lot of plugins which currently implement code coverage; however, they all use similar config, charts, and content. So it would be much better if we could have an API plugin which does the most repeated work for those plugins and offers a unified API which can be consumed by other plugins and external tools.
+
+My mentors are https://github.com/christ66[Steven Christou], https://github.com/Supun94[Supun Wanniarachchi], https://github.com/jeffpearce[Jeff Pearce] and https://github.com/oleg-nenashev[Oleg Nenashev].
+
+=== Supported Coverage Formats
+
+* Embedded
+** JaCoCo
+* Other plugins as an Extension of Code Coverage API plugin
+** Cobertura (https://github.com/jenkinsci/cobertura-plugin[Cobertura Plugin])
+** llvm-cov (https://github.com/jenkinsci/llvm-cov-plugin[llvm-cov Plugin])
+
+=== Features
+
+* Modernized coverage chart
+* Coverage trend
+* Source code navigation
+* Parallel pipeline support
+* Reports combining
+* REST API
+* Failed conditions and flexible threshold setting
+* Other small features
+
+==== Modernized Coverage Chart
+
+In the summary chart we can see the coverage summary of current coverage metric.
+image:/images/post-images/2018-06-13-code-coverage-api-plugin/summary-chart.gif[title="Summary Chart", role="center"]
+
+In the child summary chart, we can see the coverage summary of each child, also, we can use the range handler to filter item we want to see to reduce the chart size. If we want to see coverage details of the child, we can click the child name to see more information.
+image:/images/post-images/2018-06-13-code-coverage-api-plugin/child-summary-chart.gif[title="Child Summary Chart", role="center"]
+
+=== Coverage Trend
+
+We also support coverage trend to show coverage metrics changing between builds.
+image:/images/post-images/2018-06-13-code-coverage-api-plugin/trend-chart.gif[title="Trend Chart", role="center"]
+
+==== Source Code Navigation
+You can enable source code navigation by specifying Source File Storing Level to save last build source files (enable source files navigation in current and last build) or save all build source files (enable source files navigation in all builds).
+image:/images/post-images/code-coverage-api/source-files-config.png[title="Source File Navigation Configuration" role="center"]
+
+You can see source file with coverage information on File level coverage page.
+image:/images/post-images/code-coverage-api/source-files-result.png[title="Source File Navigation Result" role="center"]
+
+==== Parallel Pipeline Support
+
+We support parallel pipeline. You can call the Code Coverage API plugin in different branches like this:
+[source, groovy]
+----
+node {
+    parallel firstBranch: {
+        publishCoverage adapters: [jacoco('target/site/jacoco/jacoco.xml')]    
```

Comment:
Does it require tags? Or is it fine as is?

Reply:
We can use it in parallel execution without tag. Tag is used to combine reports.

Reply:
It would be great to add merging reports example as well

Reply:
We have merging reports example in line 76 I think.

Reply:
agreed